File: leetcode/1850-minimum-length-of-string-after-deleting-similar-ends/solution.go
func minimumLength(s string) int {
    i := 0
    j := len(s) - 1
    for j > i {
        if s[i] != s[j] {
            break
        }
        i0 := i
        for i < j && s[i] == s[i0] {
            i++
        }
        for i <= j && s[j] == s[i0] {
            j--
        }
    }
    return j - i + 1
}

File: leetcode/1850-minimum-length-of-string-after-deleting-similar-ends/solution.rs
impl Solution {
    pub fn minimum_length(s: String) -> i32 {
        let s = s.as_bytes();
        let mut i = 0;
        let mut j = s.len() - 1;
        while j > i {
            if s[i] != s[j] {
                break;
            }
            let i0 = i;
            while i < j && s[i] == s[i0] {
                i += 1;
            }
            while i <= j && s[j] == s[i0] {
                j -= 1;
            }
        }
        return j as i32 - i as i32 + 1;
    }
}
